corehq/apps/change_feed/producer/postgres.py

```diff
@@ -0,0 +1,61 @@
+import select
+import json
+import psycopg2.extensions
+from kafka import KeyedProducer
+from pillowtop.feed.interface import ChangeMeta
+
+from ..data_sources import get_document_store, POSTGRES
+from ..utils import send_to_kafka
+
+CHANNELS = {
+    XFormChannel.name: XFormChannel
+}
+
+
+class PostgresProducer(object):
+    """
+    Abstract class. Producer that pushes changes to Kafka
+    """
+
+    def __init__(self, kafka, data_source_type, data_source_name):
+        self._kafka = kafka
+        self._producer = KeyedProducer(self._kafka)
+        self._conn = get_document_store(POSTGRES)
+        self._conn.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
+        self._data_source_type = data_source_type
+        self._channel = CHANNELS[data_source_name]()
+
+    def listen(self):
+        # Each instance of PostgresProducer will take up one connection to the master database. Keep these to
+        # a minimum
+        curs = self._conn.cursor()
+        curs.execute("LISTEN {};".format(self._channel.name))
+
+        while True:
+            #  http://initd.org/psycopg/docs/advanced.html#asynchronous-notifications
+            result = select.select([self._conn], [], [], 5)
+            if result == ([], [], []):
+                print "Timeout"
+            else:
+                self._conn.poll()
+                while self._conn.notifies:
+                    notify = self._conn.notifies.pop(0)
+                    payload = json.loads(notify.payload)  # Have channel wrap this?
+
+                    change_meta = ChangeMeta(
```

Comment:
whoops you're right, i got type and name mixed up. 

regardless, it should be fine to change/extend those concepts as we see fit. I was originally thinking that `data_source_name` would be a unique identifier of the sql database that you could use to get a connection string. since I didn't actually implement a sql backend yet it didn't occur to me that unlike couch you would also need a `table` to pull something out of a sql db but of course that makes sense. whether we should just infer that from the `doc_type` for ease of backwards compatibility or create a new construct for it doesn't seem all that important to me right now as long as we can get something that readily works with both backends. 

in general as long as there's enough information in the `ChangeMeta` to uniquely tell the consumer how to get the source data back out of the DB that's all that matters. I figured we'd just iterate on is as we wrote the first sql backend (so the exercise we're going through here is good/expected).